backend/graph/tools.py

```
@@ -21,21 +23,34 @@ def vectordb_search(query: str, config: RunnableConfig) -> str:
     except Exception:
         logging.exception("vectordb_search failed")
         return "[Search error: the document search is temporarily unavailable.]"
+    
+    # Filter sources with LLM
+    def check_chunk(c):
+        return is_relevant_source(llm, query, c.page_content).is_relevant
 
+    with ThreadPoolExecutor() as executor:
+        relevance = list(executor.map(check_chunk, chunks))
+
+    chunks = [c for c, ok in zip(chunks, relevance) if ok]
```

Ruff (0.15.12)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)